### PR TITLE
When on the levelbuilder branch, restrict changes to dashboard/config…

### PR DESCRIPTION
**Pull request by mehalshah**
…/levels directory.

Gets us on the path to better seperating parts of our pipeline.

### CODE REVIEW

File: tools/hooks/hooks_utils.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class HooksUtils
+  def self.get_modified_files
+    Dir.chdir REPO_DIR
```
Comment (joshlory) [inline tools/hooks/hooks_utils.rb]: This method relies on the caller setting `REPO_DIR`.

File: tools/hooks/restrict_levelbuilder_changes.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require_relative 'hooks_utils.rb'
+
+REPO_DIR = File.expand_path('../../../', __FILE__)
+LEVELS_DIR = File.expand_path(REPO_DIR + '/dashboard/config/levels', __FILE__)
```
Comment (joshlory) [inline tools/hooks/restrict_levelbuilder_changes.rb]: `/dashboard/config/levels` -> `/dashboard/config/scripts`

File: tools/hooks/restrict_levelbuilder_changes.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+Dir.chdir REPO_DIR
+branchname = `git rev-parse --abbrev-ref HEAD`.strip
+
+exit(0) unless branchname == 'levelbuilder'
```
Comment (joshlory) [inline tools/hooks/restrict_levelbuilder_changes.rb]: **If** `branchname == 'levelbuilder'`?  Condition seems to be inverted.

File: tools/hooks/restrict_levelbuilder_changes.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+modified_files = HooksUtils.get_modified_files
+
+modified_files.each do |filename|
+  raise 'Levelbuilder branch should only commit files in levels directory' if !(filename.start_with? LEVELS_DIR)
```
Comment (joshlory) [inline tools/hooks/restrict_levelbuilder_changes.rb]: Nit: `if !` -> `unless filename.start_with? LEVELS_DIR`.

File: tools/hooks/restrict_levelbuilder_changes.rb (hunk 3 above)
Comment (joshlory) [inline tools/hooks/restrict_levelbuilder_changes.rb]: Never mind, I misunderstood what this hook was checking.

File: tools/hooks/restrict_levelbuilder_changes.rb (hunk 2 above)
Comment (mehalshah) [inline tools/hooks/restrict_levelbuilder_changes.rb]: Changed

File: tools/hooks/hooks_utils.rb (hunk 1 above)
Comment (mehalshah) [inline tools/hooks/hooks_utils.rb]: Moved definition into hooks_utils

**Comment by joshlory** [comment]
LGTM

**Pull request merged**